File: CCCInventory/ClientApp/src/app/models/order.ts
import { Time } from "@angular/common";
import { Cake } from "./cake";
import { Cookie } from "./cookie";
import { Cupcake } from "./cupcake";
import { Pupcake } from "./pupcake";

export class Order {
  orderNumber: number | undefined;
  orderDate: Date | undefined;
  pickupTime: Time | undefined;
  deliveryLocation?: string;
  custName: string | undefined;
  custEmail: string | undefined;
  custPhone: string | undefined;
  formFiles?: Array<File>;
  details: string | undefined;
  pickupOrDelivery: boolean | undefined;
  secondaryName?: string;
  secondaryPhone?: string;
  initialContact?: string;
  contractSent?: boolean | false;
  dayOfTextSent?: boolean | false;
  confirmationTextSent?: boolean | false;
  cakes?: Array<Cake>;
  cupcakes?: Array<Cupcake>;
  pupcakes?: Array<Pupcake>;
  cookies?: Array<Cookie>;
  totalCost?: number;
  depositAmount?: number;
  depositPaymentMethod?: string;
  depositDateTime?: Date;
  finalPaymentMethod?: string;
  finalPaymentDateTime?: Date;
  dateOrderPlaced: Date | undefined;
  paidInFull?: boolean | undefined;
}

File: CCCInventory/Order.cs
using System.ComponentModel.DataAnnotations;

namespace CCCInventory
{
    public class Order
    {
        [Key]
        public int OrderNumber { get; set; }
        public DateTime? OrderDateTime { get; set; }
        public string? DeliveryLocation { get; set; }
        public string? CustName { get; set; }
        public string? CustEmail { get; set; }
        public string? CustPhone { get; set; }
        List<string>? FormFiles { get; set; }
        public string? Details { get; set; }
        public bool? PickupOrDelivery { get; set; }
        public string? SecondaryName { get; set; }
        public string? SecondaryPhone { get; set; }
        public string? InitialContact { get; set; }
        public bool ContractSent { get; set; } = false;
        public bool DayOfTextSent { get; set; } = false;
        public bool ConfirmationTextSent { get; set; } = false;
        public List<Cake>? Cakes { get; set; }
        public List<Cupcake>? Cupcakes { get; set; }
        public List<Pupcake>? Pupcakes { get; set; }
        public List<Cookie>? Cookies { get; set; }
        public double? TotalCost { get; set; }
        public double? DepositAmount { get; set; }
        public string? DepositPaymentMethod { get; set; }
        public DateTime? DepositDateTime { get; set; }
        public string? FinalPaymentMethod { get; set; }
        public DateTime? FinalPaymentDateTime { get; set; }
        public DateTime DateOrderPlaced { get; set; }
        public bool? PaidInFull { get; set; }
        public bool DeleteFlag { get; set; } = false;
    }
}
